api/v1alpha2/agentpool_types.go

```diff
@@ -31,6 +32,11 @@ type AgentToken struct {
 	LastUsedAt *int64 `json:"lastUsedAt,omitempty"`
 }
 
+type AgentDeployment struct {
+	Replicas *int32     `json:"replicas,omitempty"`
+	Spec     v1.PodSpec `json:"spec"`
```

Comment:
I see the logic in just having them be able to configure a PodSpec for the agent here, but I'm wondering if this introduces too much complexity to this interface – PodSpec adds a lot of surface area. Maybe we could simplify it down to just `Container`, or even simpler just allowing them to optionally configure the image?

I fee like the value here is that the operator is going to take care of all the details of the deployment of the agent and the user really just has to care about how many replicas they need, and potentially using a different image.

Reply:
Yeah, I did ponder quite a bit about how much surface area to expose here. There's a lot of good arguments for starting out with a small set of options. But the more I thought about what might be needed to run the agent IRL, the more I think we should be future proof here, given how tedious CRD schema changes can be (when done correctly). I can imagine how users might soon ask for allowing volumes to be attached to the agent pods, resource quotas (to fit the workers on busy / small clusters), or various types of credentials that need injecting into the agent for real-life work with TF configs. Addresing all these asks reactively by adding / changing attributes in a CRD the cannonical way requires issuing a new API version while ensuring backwards compatibility for all the earlier versions (I'm thinking migration webhooks).

OTH, even though we would allow for a vanilla PodSpec in there, the complexity is not exposed to the users unless they need it. The minimum set of inputs is the replica count and we can default the container image to the official one if not provided by the user.

That was my train of thought for this choice. Is there any aspect of complexity that you are concerned about which I didn't capture here?